### PR TITLE
[Impeller] Refactor CapabilitiesGLES into a Capabilties.

### PR DESCRIPTION
**Pull request by jonahwilliams**
From https://github.com/flutter/engine/pull/46585

For Framebuffer fetch and other changes we should refactor the CapabilitiesGLES into a regular capabilities.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Doing this separately so that changes that require other extension checks aren't blocked on the framebuffer fetch PR.

File: impeller/renderer/backend/gles/context_gles.cc [hunk 1]
```diff
@@ -58,26 +59,7 @@ ContextGLES::ContextGLES(std::unique_ptr<ProcTableGLES> gl,
...
-            .SetDefaultDepthStencilFormat(PixelFormat::kD24UnormS8Uint)
-            .SetSupportsCompute(false)
-            .SetSupportsComputeSubgroups(false)
-            .SetSupportsReadFromResolve(false)
-            .SetSupportsReadFromOnscreenTexture(false)
-            .SetSupportsDecalSamplerAddressMode(false)
-            .SetSupportsDeviceTransientTextures(false)
-            .Build();
-  }
+  { device_capabilities_ = reactor_->GetProcTable().GetCapabilities(); }
```
Comment (bdero) [inline impeller/renderer/backend/gles/context_gles.cc:62]: Nit: No need for these curlys?

**Review by bdero — APPROVED** [review]
LGTM

File: impeller/renderer/backend/gles/context_gles.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/gles/context_gles.cc]: Removed.

File: impeller/renderer/backend/gles/context_gles.cc [hunk 2]
```diff
@@ -58,27 +59,7 @@ ContextGLES::ContextGLES(std::unique_ptr<ProcTableGLES> gl,
         std::shared_ptr<SamplerLibraryGLES>(new SamplerLibraryGLES());
   }
 
-  // Create the device capabilities.
-  {
-    device_capabilities_ =
-        CapabilitiesBuilder()
```
Comment (chinmaygarde) [inline impeller/renderer/backend/gles/context_gles.cc:64]: Can we get rid of CapabilitiesBuilder entirely? All backends should follow the pattern of having a custom capabilities subclass.

**Pull request merged**